syntax = "proto3";

service PasswordManager {
  rpc StorePassword (PasswordRequest) returns (PasswordResponse);
  rpc RetrievePassword (AccountRequest) returns (PasswordResponse);
  rpc DeletePassword (AccountRequest) returns (StatusResponse);
  rpc ListAccounts (Empty) returns (AccountListResponse);
}

message PasswordRequest {
  string account = 1;
  string password = 2;
}

message PasswordResponse {
  string account = 1;
  string password = 2;
  string status = 3;
}

message AccountRequest {
  string account = 1;
}

message AccountListResponse {
  repeated string accounts = 1;
}

message StatusResponse {
  string status = 1;
}

message Empty {}
